### PR TITLE
For #4637 - Query manifests with share targets

### PR DESCRIPTION
**Pull request by NotWoods**
My intent is to expose this function so that we can use it with the Fenix custom share sheet.

We want to support sharing from websites to PWAs. PWAs can define a "share target" which will receive sharing data. We can expose PWAs in the custom share sheet so that other websites can send data to them.

---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/pwa/src/main/java/mozilla/components/feature/pwa/db/ManifestDatabase.kt [hunk 1]
```diff
@@ -42,6 +43,21 @@ internal abstract class ManifestDatabase : RoomDatabase() {
             }
         }
 
+        @VisibleForTesting
+        internal val MIGRATION_2_3: Migration = object : Migration(2, 3) {
+            override fun migrate(database: SupportSQLiteDatabase) {
+                val cursor = database.query("SELECT * FROM manifests LIMIT 0,1")
+
+                if (cursor.getColumnIndex("share_targets") < 0) {
```
Comment (grigoryk) [inline components/feature/pwa/src/main/java/mozilla/components/feature/pwa/db/ManifestDatabase.kt:51]: Why do you need to do this? You know it's a migration for `2` to `3`, and this column wasn't present on `2`.

Will this code ever run in a non-migration scenario..?

File: components/feature/pwa/src/main/java/mozilla/components/feature/pwa/db/ManifestDao.kt [hunk 2]
```diff
@@ -28,6 +28,10 @@ internal interface ManifestDao {
     @Query("SELECT count(start_url) from manifests WHERE :url LIKE scope||'%' AND used_at < :deadline")
     fun hasRecentManifest(url: String, deadline: Long): Int
 
+    @WorkerThread
```
[the remark this answers is not in the document]
Comment (NotWoods) [reply components/feature/pwa/src/main/java/mozilla/components/feature/pwa/db/ManifestDao.kt:31]: That requires a newer Room version. Upgrading causes some build issues so we're currently pinned to 2.2.5.

**Comment by grigoryk** [comment]
Looks good, land whenever you want.

**Comment by NotWoods** [comment]
bors r=grigoryk

**Pull request merged**